Connect to NATS cluster with username/password

In the [documentation of NATS server](https://nats-io.github.io/docs/developer/security/userpass.html#connecting-with-a-userpassword-in-the-url) I found that you can connect to a NATS server with the Go NATS client with a username and password in the url. When I use a username and password in the NATS server url for Liftbridge I get the following message:
```
panic: failed to connect to NATS: parse nats://myuser:mypassword: invalid port ":mypassword" after host
```
(I have changed the username and password with placeholders, the actual password has a special character in it (`^`))

Should I connect in some other way to use a username/password for the NATS cluster, or is this not possible yet?

--- a/documentation/configuration.md
+++ b/documentation/configuration.md
@@ -126,6 +126,8 @@ the configuration file.
 | Name | Flag | Description | Type | Default | Valid Values |
 |:----|:----|:----|:----|:----|:----|
 | servers | nats-servers | List of NATS hosts to connect to. | list | nats://localhost:4222 | |
+| user | | Username to use to connect to NATS servers. | string | | |
+| password | | Password to use to connect to NATS servers. | string | | |
 
 ### Log Configuration Settings
 

--- a/server/config.go
+++ b/server/config.go
@@ -295,6 +295,12 @@ func parseNATSConfig(m map[string]interface{}, opts *nats.Options) error {
 			for i, p := range servers {
 				opts.Servers[i] = p.(string)
 			}
+		case "user":
+			user := v.(string)
+			opts.User = user
+		case "password":
+			password := v.(string)
+			opts.Password = password
 		default:
 			return fmt.Errorf("Unknown nats configuration setting %q", k)
 		}

--- a/server/config_test.go
+++ b/server/config_test.go
@@ -1,0 +1,15 @@
+package server
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// Ensure we can properly parse NATS username and password from a config file.
+func TestConfigParseNATSAuth(t *testing.T) {
+	config, err := NewConfig("configs/nats_auth.conf")
+	require.NoError(t, err)
+	require.Equal(t, "admin", config.NATS.User)
+	require.Equal(t, "password", config.NATS.Password)
+}

--- a/server/configs/nats_auth.conf
+++ b/server/configs/nats_auth.conf
@@ -1,0 +1,4 @@
+nats {
+    user: "admin"
+    password: "password"
+}

